opentelemetry-otlp/src/exporter/http/mod.rs

```diff
@@ -50,6 +49,9 @@ pub struct HttpConfig {
 
     /// Additional headers to send to the collector.
     headers: Option<HashMap<String, String>>,
+
+    /// The compression algorithm to use when communicating with the collector.
```

Comment:
```suggestion
    /// The compression algorithm to use when communicating with the OTLP endpoint.
```

Reply:
There's a lot of 'unnecessary concretisation' in this fashion scattered around; i've just fixed this and the obvious adjacent ones